src/cogent3/core/new_alignment.py

```diff
@@ -912,6 +1007,378 @@ def dotplot(
+        names
+            returns only scores for these sequences and in the name order
+        ui
+            argument to control the display of a progress bar. i.e.,
+            'show_progress=True'
+
+
+        Returns
+        -------
+        numpy array of log2 based scores at every position
+        """
+        assert not self.is_ragged(), "all sequences must have same length"
+        from cogent3.parse import cisbp, jaspar
+
+        assert pssm or path, "Must specify a PSSM or a path"
+        assert not (pssm and path), "Can only specify one of pssm, path"
+
+        if isinstance(names, str):
+            names = [names]
+
+        if path:
+            is_cisbp = path.endswith("cisbp")
+            is_jaspar = path.endswith("jaspar")
+            if not (is_cisbp or is_jaspar):
+                raise NotImplementedError(f"Unknown format {path.split('.')[-1]}")
+
+            if is_cisbp:
+                pfp = cisbp.read(path)
+                pssm = pfp.to_pssm(background=background)
+            else:
+                _, pwm = jaspar.read(path)
+                pssm = pwm.to_pssm(background=background, pseudocount=pseudocount)
+
+        assert isinstance(pssm, PSSM)
+        array_align = hasattr(self, "array_seqs")
+        assert set(pssm.motifs) == set(self.moltype)
+        if array_align and list(pssm.motifs) == list(self.moltype):
+            if names:
+                name_indices = [self.names.index(n) for n in names]
+                data = self.array_seqs.take(name_indices, axis=0)
+            else:
+                data = self.array_seqs
+
+            result = [pssm.score_indexed_seq(seq) for seq in ui.series(data)]
+        else:
+            seqs = [self.seqs[n] for n in names] if names else self.seqs
+            result = [pssm.score_seq(seq) for seq in ui.series(seqs)]
+
+        return numpy.array(result)
+
+    def get_ambiguous_positions(self):
+        """Returns dict of seq:{position:char} for ambiguous chars.
+
+        Used in likelihood calculations.
+        """
+        result = {}
+        for name in self.names:
+            result[name] = ambig = {}
+            for i, motif in enumerate(self.seqs[name]):
+                if self.moltype.is_ambiguity(motif):
+                    ambig[i] = motif
+        return result
+
+    def trim_stop_codons(self, gc: Any = 1, strict: bool = False):
+        """Removes any terminal stop codons from the sequences
+
+        Parameters
+        ----------
+        gc
+            valid input to cogent3.get_code(), a genetic code object, number
+            or name, defaults to standard code
+        strict
+            If True, raises an exception if a seq length not divisible by 3
+        """
+        if not self.has_terminal_stop(gc=gc, strict=strict):
+            return self
+
+        new_seqs = {s.name: s.trim_stop_codon(gc=gc, strict=strict) for s in self.seqs}
+        result = make_unaligned_seqs(new_seqs, moltype=self.moltype, info=self.info)
```

Comment:
another use of make_unaligned_seqs(). Make a comment about refactor to directly call `SequenceCollection` class.